src/box/lua/console.c

```diff
@@ -311,7 +314,7 @@ console_can_hide_show_prompt(void)
 static void
 console_hide_prompt(void)
 {
-	if (!console_can_hide_show_prompt())
+	if (!console_can_hide_show_prompt() || !cord_is_main())
```

Comment:
I looks important to do this check from inside the callback (as you implemented it). Otherwise one thread may set a callback and another will call it. I would leave a short note here. Just to don't step into the problem later.

Reply:
> Otherwise one thread may set a callback and another will call it. 

TBH, I don't understand what you mean. Could you please elaborate and provide an example?

Reply:
My first question was 'why don't do this in `luaT_console_setup_write_cb()` and `luaT_console_cleanup_write_cb()`?' But if we would go this way, we would get the situation described above.

Reply:
Now I get it... 

But AFAIC, only the main thread can set up callbacks? Actually, I think it would be better to check for this in the setup functions.

@locker RFC

Reply:
> But AFAIC, only the main thread can set up callbacks?

I guess so. But when the callback is set from the main thread, another thread may call it. So I think that your current implementation is the right one.

Reply:
Yeah, I just wrote some nonsense, sorry. Tuesday evening… Well then I guess it’s pretty intuitive that the check needs to be done in the callback itself. If you don’t mind, I would skip the comment, as I can’t come with one, seems obvious at first glance.

Reply:
Okay.